### ISSUE TITLE
Download images eventhough disabled in settings

### ISSUE THREAD

**Issue by MaxMuehlbauer** [Foundry376/Mailspring#1071]
##### Are there any related issues?
not found

##### What operating system are you using?
Windows 10 Pro 64-Bit

##### What version of Mailspring are you using?
v1.4.2-f587b7b7

--

### **Although automatically loading images is disabled, sometimes Mailspring downloads them**

This occurs from time to time. For me this is a **major security issue**, because I don't want the sender to be able to track me just through opening the mail. It could potentially be used for spam or worse.

In the settings I have disabled "Automatically load images in viewed messages".

See the screenshot:
![mailspring-imageloading](https://user-images.githubusercontent.com/18306885/45507466-82a8aa00-b792-11e8-91b9-18ba77355e50.PNG)

And here's the code for the image part:
```
<!--HEADER-->
<table cellpadding="0" cellspacing="0" border="0" width="600" class="full-width-content">
	<tr>
  	<td align="center" style="background-color:#1e252b;">
    	<!--content block-->
    	<table cellpadding="0" cellspacing="0" border="0" width="100%">
        <tr>
          <!--logo-->
          <td align="left">
            <a href="#" target="_blank" style="border:none;">
            <span class="headerImage"></span>
            <img src="http://newsletters-cdn.ea.com/origin/rtmtest2/images/header_oa.jpg" border="0" style="display:block;" alt="" class="hide"></a>
          </td>
          <!--end logo-->                
        </tr>
      </table>
      <!--end content block-->	
    </td>
  </tr>
</table>
<!--end HEADER-->
```
I know, the image source doesn't include tracking parameters. But I am unsure whether Mailspring is only doing this, when it does not include tracking parameters. But than again - how would Mailspring know what parameters are for tracking? And still, the setting I disabled clearly does not include an exceptional case like this.

##### Is this an email sync or connection issue?
no

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
no

##### Is the issue reproducible with a particular attachment, message, signature, etc?
Maybe it's because sender (noreply@e.ea.com) and reply-to (reply-xyz@e.ea.com) differ. But that's just a guess and a bad guess on top 🤷‍♂️

**Comment by bengotow** [comment]
Wow this is super interesting. Offhand, I don't see any reason this image wouldn't be blocked. I'll see if I can get to the bottom of this.

**Comment by derrickontiveros** [comment]
Any update on this? I've submitted several support tickets with no response.

**Comment by bengotow** [comment]
Hey folks! This has been fixed in the upcoming release - it turned out to be super tricky to track down but I finally got it. Our implementation uses a regular expression to wipe all src="", background="", CSS backgrounds / images, etc. from message bodies when this filtering is enabled, but it turns out Javascript's "Regexp.test()" method is stateful, meaning it internally iterates through matches and eventually returns false, even if the test string matches. This would cause Mailspring to randomly and occasionally skip image filtering if a message was re-rendered a few times.

bc25ebfe9e8407298a689578bb14300ae94bb9fe and 250f88004069fd02e3a1b7399093364562740506.

I also made it so that images embedded within the message itself as inline attachments are also hidden if you choose not to view images, since those were confusing (even though they didn't load over the network, still preferable to hide them I think). I also disabled the loading of external CSS stylesheets and fonts, which theoretically could also be used for tracking.

**Comment by bengotow** [comment]
Hey folks, thanks for reporting this. Happy to say this has been resolved in 1.7.6 which shipped this afternoon. 🚀  The app should auto-update, but if it doesn't you can also grab the new release from GitHub's Releases tab or from https://getmailspring.com/download. Thanks for your patience and help tracking this down, enjoy the update!
